modin/experimental/backends/omnisci/query_compiler.py

```diff
@@ -517,9 +517,15 @@ def setitem(self, axis, key, value):
         if axis == 1 or not isinstance(value, type(self)):
             return super().setitem(axis=axis, key=key, value=value)
 
-        return self._setitem(axis, key, value)
+        try:
+            result = self._setitem(axis, key, value)
+        # OmniSci engine does not yet support cases when `value` is not a subframe of `self`.
```

Comment:
There is no explicit check for whether `value` is a subframe or not?

Reply:
Engine check it inside and raises `NotImplementedError` if it is not a subframe, in this case we're catching this exception and defaulting to pandas

Reply:
Technically we can do this explicit check at the query compiler level by calling `modin_frame._find_common_projection_base` (this is how the engine determines if the other frame is a subframe of `self`), but this method is linear on the length of operation tree and gives us some overhead of inserting because we will call this method twice: first in query compiler and then in the engine.

discussed it with @ienkovich and agreed on an approach with catching exception

Reply:
ok